haystack/nodes/retriever/_openai_encoder.py

```diff
@@ -94,37 +60,29 @@ def _ensure_text_limit(self, text: str) -> str:
         Ensure that length of the text is within the maximum length of the model.
         OpenAI v1 embedding models have a limit of 2046 tokens, and v2 models have a limit of 8191 tokens.
         """
+        n_tokens = count_openai_tokens(text, self._tokenizer)
+        if n_tokens <= self.max_seq_len:
```

Comment:
I think it's a bit confusing how we use many different terms for number of tokens:
`n_tokens`, `max_seq_len`, `max_tokens`, `max_tokens_limit` and earlier `leftover_token_len`, `n_full_prompt`, `n_full_prompt_tokens `, `n_skipped_tokens`. Maybe we could follow a convention of using `n_` when we count tokens and `max_` when we set a limit? `_len` I would leave out. So `leftover_token_len` could become `n_leftover_tokens` and `max_seq_len` could become `max_tokens`. What do you think?

Reply:
I agree that this is confusing and not consistent. Maybe we could combine this with your previous comment into one new PR to handle naming conventions?